packages/smooth_app/lib/helpers/product_cards_helper.dart

```diff
@@ -9,6 +9,23 @@ import 'package:smooth_app/generic_lib/widgets/smooth_card.dart';
 String getProductName(Product product, AppLocalizations appLocalizations) =>
     product.productName ?? appLocalizations.unknownProductName;
 
+String getProductBrands(Product product, AppLocalizations appLocalizations) {
+  final String? brands = product.brands;
+  if (brands == null) {
+    return appLocalizations.unknownBrand;
+  } else {
+    return formatProductBrands(brands, appLocalizations);
+  }
+}
+
+/// Correctly format word separators between words (e.g. comma in English)
+String formatProductBrands(String brands, AppLocalizations appLocalizations) {
+  final String separator = appLocalizations.word_separator;
+  final String separatorChar = appLocalizations.word_separator_char;
+  final RegExp regex = RegExp('\\s*$separatorChar\\s*');
+  return brands.replaceAll(regex, separator);
```

Comment:
Your code is error-proof.
Let's say a translator indicates that `word_separator_char` is `*`, then your Regex will never work.